### ISSUE TITLE
User stylesheets, UA stylesheets - still something we expect to be real?

### ISSUE THREAD

**Issue by svgeesus** [w3c/csswg-drafts#2237]
CSS 2.2 section [3.2 UA conformance](https://drafts.csswg.org/css2/conform.html#conformance):

> UAs must allow users to specify a file that contains the user style sheet. UAs that run on devices without any means of writing or specifying files are exempted from this requirement. Additionally, UAs may offer other means to specify user preferences, for example, through a GUI. 
> https://drafts.csswg.org/css2/conform.html#conformance

Microsoft Edge does not support User Stylesheets. 
https://msdn.microsoft.com/en-us/library/hh674753(v=vs.85).aspx

Chrome removed User stylesheets ages ago (chrome 33)
https://bugs.chromium.org/p/chromium/issues/detail?id=347016
https://news.ycombinator.com/item?id=7329855

The conformance section does not mention UA default stylesheets at all, but they are listed in section [6.4 the cascade](https://drafts.csswg.org/css2/cascade.html#cascade):

> User agent: Conforming user agents must apply a default style sheet (or behave as if they did). A user agent's default style sheet should present the elements of the document language in ways that satisfy general presentation expectations for the document language (e.g., for visual browsers, the EM element in HTML is presented using an italic font). See A sample style sheet for HTML for a recommended default style sheet for HTML documents. 

I came across this again recently in the context of test for Fonts 3. A couple of tests require a particular font to be set in the UA stylesheet. For example [font-052](https://test.csswg.org/harness/test/css-fonts-3_dev/font-052/). Everyone fails that test, and it has less results than other tests (skipping is easier than finding how to change the UA stylesheet and restarting the browser, then changing it back again for one test).

Do we want to still 
a) say that UA stylesheets are a thing
b) insist that they be modifiable by users (as a stylesheet, not as prefs in a dialog box)

If not, then CSS2 should be modified.

if we do, then we need browser vendors on board to expose UA styles to user editing. Otherwise, how can we pass tests like these?

**Comment by svgeesus** [comment]
User stylesheets (not UA stylesheets, I think) are also a WAI UA Accessibility Guidelines requirement
https://www.w3.org/TR/UAAG20/#gl-style-sheets-config

**Comment by emilio** [comment]
Also, Blink was adding them back last time I checked, see https://bugs.chromium.org/p/chromium/issues/detail?id=632009.

**Comment by svgeesus** [comment]
Great link! Interesting to see that extensions rely on injecting user stylesheets.

**Comment by fantasai** [comment]
There's no requirement that UA stylesheets be modifiable by users; in fact they are most definitely not intended to be modifiable by users--the users should be modifying the user stylesheet. There's also no requirement that the UA represent the user stylesheet as CSS syntax--if it wants to inject rules directly into the cascade internally, that's fine and allowable. So I'm not sure what the problem is that you're asking us to fix.

**Comment by fantasai** [comment]
I also don't see why those two tests need the UA's default font to be set to anything? The test should just specify `div { font-family: whatever-the-default-was-supposed-to-be }` if it needs a specific font to be specified earlier in the cascade.

**Comment by fantasai** [comment]
Edit: Nevermind on sentence 2-3 in the earlier comment. It looks like 3.2 and 6.4 conflict.

**Comment by Nadya678** [comment]
@fantasai: But due to possibility of tracking users, I propose Helvetica, Arial, Liberation Sans 16px as default font. Or other with character width compatible with Arial/Helvetica.

**Comment by svgeesus** [comment]
No, it needs to be the specific font required by the test. Helvetica etc as you suggest would not suit the test purpose.

**Comment by svgeesus** [comment]
>  The test should just specify div { font-family: whatever-the-default-was-supposed-to-be } if it needs a specific font to be specified earlier in the cascade

That fix works fine to solve my specific question, I will patch those two tests. The more general question remains open.
